### ISSUE TITLE
2.9.1: sphinx warnings

### ISSUE THREAD

**Issue by kloczek** [python-babel/babel#801]
Looks like new sphinx 4.x shows some warning on generate documentation
```console
+ /usr/bin/python3 setup.py build_sphinx -b man --build-dir build/sphinx
running build_sphinx
Running Sphinx v4.1.2
making output directory... done
loading intersphinx inventory from https://docs.python.org/2/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: [new config] 27 added, 0 changed, 0 removed
reading sources... [100%] support
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/babel/messages/extract.py:docstring of babel.messages.extract.extract_javascript:13: WARNING: Unexpected indentation.
CHANGES:237: WARNING: Title underline too short.

Minor Improvements and bugfixes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CHANGES:237: WARNING: Title underline too short.

Minor Improvements and bugfixes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CHANGES:348: WARNING: Inline emphasis start-string without end-string.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... python-babel.3 { intro installation locale dates numbers messages cmdline setup support api/index api/core api/dates api/languages api/lists api/messages/index api/messages/catalog api/messages/extract api/messages/mofile api/messages/pofile api/numbers api/plural api/support api/units dev changelog license } /home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
done
build succeeded, 13 warnings.
```

**Comment by kloczek** [comment]
After switching to pep517 build I'm no longer able to build man page

```console
+ /usr/bin/python3 -sBm build -w --no-isolation -C=import_cldr
* Getting dependencies for wheel...
running egg_info
creating Babel.egg-info
writing manifest file 'Babel.egg-info/SOURCES.txt'
warning: no files found matching 'babel/global.dat'
warning: no files found matching 'babel/locale-data/*.dat'
warning: no previously-included files matching '*' found under directory 'docs/_build'
warning: no previously-included files matching '*.pyc' found under directory 'tests'
warning: no previously-included files matching '*.pyo' found under directory 'tests'
writing manifest file 'Babel.egg-info/SOURCES.txt'
* Building wheel...
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/babel
copying babel/__init__.py -> build/lib/babel
copying babel/_compat.py -> build/lib/babel
copying babel/core.py -> build/lib/babel
copying babel/dates.py -> build/lib/babel
copying babel/languages.py -> build/lib/babel
copying babel/lists.py -> build/lib/babel
copying babel/localedata.py -> build/lib/babel
copying babel/numbers.py -> build/lib/babel
copying babel/plural.py -> build/lib/babel
copying babel/support.py -> build/lib/babel
copying babel/units.py -> build/lib/babel
copying babel/util.py -> build/lib/babel
creating build/lib/babel/messages
copying babel/messages/__init__.py -> build/lib/babel/messages
copying babel/messages/catalog.py -> build/lib/babel/messages
copying babel/messages/checkers.py -> build/lib/babel/messages
copying babel/messages/extract.py -> build/lib/babel/messages
copying babel/messages/frontend.py -> build/lib/babel/messages
copying babel/messages/jslexer.py -> build/lib/babel/messages
copying babel/messages/mofile.py -> build/lib/babel/messages
copying babel/messages/plurals.py -> build/lib/babel/messages
copying babel/messages/pofile.py -> build/lib/babel/messages
creating build/lib/babel/localtime
copying babel/localtime/__init__.py -> build/lib/babel/localtime
copying babel/localtime/_unix.py -> build/lib/babel/localtime
copying babel/localtime/_win32.py -> build/lib/babel/localtime
running egg_info
warning: no files found matching 'babel/global.dat'
warning: no files found matching 'babel/locale-data/*.dat'
warning: no previously-included files matching '*' found under directory 'docs/_build'
warning: no previously-included files matching '*.pyc' found under directory 'tests'
warning: no previously-included files matching '*.pyo' found under directory 'tests'
writing manifest file 'Babel.egg-info/SOURCES.txt'
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/babel
creating build/bdist.linux-x86_64/wheel/babel/messages
creating build/bdist.linux-x86_64/wheel/babel/localtime
running install_egg_info
Copying Babel.egg-info to build/bdist.linux-x86_64/wheel/Babel-2.9.1-py3.8.egg-info
running install_scripts
Successfully built Babel-2.9.1-py2.py3-none-any.whl
+ CFLAGS='-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none'
+ CXXFLAGS='-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none'
+ FFLAGS='-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -I/usr/lib64/gfortran/modules'
+ FCFLAGS='-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -I/usr/lib64/gfortran/modules'
+ LDFLAGS='-Wl,-z,relro -Wl,--as-needed -Wl,--gc-sections -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto=auto -flto-partition=none -fuse-linker-plugin -Wl,--build-id=sha1'
+ CC=/usr/bin/gcc
+ CXX=/usr/bin/g++
+ FC=/usr/bin/gfortran
+ AR=/usr/bin/gcc-ar
+ NM=/usr/bin/gcc-nm
+ RANLIB=/usr/bin/gcc-ranlib
+ export CFLAGS CXXFLAGS FFLAGS FCFLAGS LDFLAGS CC CXX FC AR NM RANLIB
+ PBR_VERSION=2.9.1
+ SETUPTOOLS_SCM_PRETEND_VERSION=2.9.1
+ /usr/bin/python3 -sBm build -w --no-isolation
* Getting dependencies for wheel...
running egg_info
warning: no files found matching 'babel/global.dat'
warning: no files found matching 'babel/locale-data/*.dat'
warning: no previously-included files matching '*' found under directory 'docs/_build'
warning: no previously-included files matching '*.pyc' found under directory 'tests'
warning: no previously-included files matching '*.pyo' found under directory 'tests'
writing manifest file 'Babel.egg-info/SOURCES.txt'
* Building wheel...
running bdist_wheel
running build
running build_py
running egg_info
warning: no files found matching 'babel/global.dat'
warning: no files found matching 'babel/locale-data/*.dat'
warning: no previously-included files matching '*' found under directory 'docs/_build'
warning: no previously-included files matching '*.pyc' found under directory 'tests'
warning: no previously-included files matching '*.pyo' found under directory 'tests'
writing manifest file 'Babel.egg-info/SOURCES.txt'
running install
running install_lib
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/babel
creating build/bdist.linux-x86_64/wheel/babel/messages
creating build/bdist.linux-x86_64/wheel/babel/localtime
running install_egg_info
Copying Babel.egg-info to build/bdist.linux-x86_64/wheel/Babel-2.9.1-py3.8.egg-info
running install_scripts
Successfully built Babel-2.9.1-py2.py3-none-any.whl
+ /usr/bin/python3 setup.py build_sphinx -b man --build-dir build/sphinx
running build_sphinx
Running Sphinx v4.3.2
making output directory... done
loading intersphinx inventory from https://docs.python.org/2/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: [new config] 27 added, 0 changed, 0 removed
reading sources... [100%] support
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/babel/messages/extract.py:docstring of babel.messages.extract.extract_javascript:13: WARNING: Unexpected indentation.
CHANGES:237: WARNING: Title underline too short.

Minor Improvements and bugfixes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CHANGES:237: WARNING: Title underline too short.

Minor Improvements and bugfixes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CHANGES:348: WARNING: Inline emphasis start-string without end-string.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... python-babel.3 { intro installation locale dates numbers messages cmdline setup support api/index api/core api/dates api/languages api/lists api/messages/index api/messages/catalog api/messages/extract api/messages/mofile api/messages/pofile api/numbers api/plural api/support api/units dev changelog license } failed

Exception occurred:
  File "/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/babel/core.py", line 27, in _raise_no_data_error
    raise RuntimeError('The babel data files are not available. '
RuntimeError: The babel data files are not available. This usually happens because you are using a source checkout from Babel and you did not build the data files.  Just make sure to run "python setup.py import_cldr" before installing the library.
The full traceback has been saved in /tmp/sphinx-err-w8mv3lci.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```
```console
[tkloczko@ss-desktop SPECS]$ cat /tmp/sphinx-err-w8mv3lci.log
# Sphinx version: 4.3.2
# Python version: 3.8.12 (CPython)
# Docutils version: 0.17.1 release
# Jinja2 version: 3.0.3
# Last messages:
#   api/messages/pofile
#   api/numbers
#   api/plural
#   api/support
#   api/units
#   dev
#   changelog
#   license
#   }
#   failed
# Loaded extensions:
#   sphinx.ext.mathjax (4.3.2) from /usr/lib/python3.8/site-packages/sphinx/ext/mathjax.py
#   sphinxcontrib.applehelp (1.0.2) from /usr/lib/python3.8/site-packages/sphinxcontrib/applehelp/__init__.py
#   sphinxcontrib.devhelp (1.0.2) from /usr/lib/python3.8/site-packages/sphinxcontrib/devhelp/__init__.py
#   sphinxcontrib.htmlhelp (2.0.0) from /usr/lib/python3.8/site-packages/sphinxcontrib/htmlhelp/__init__.py
#   sphinxcontrib.serializinghtml (1.1.5) from /usr/lib/python3.8/site-packages/sphinxcontrib/serializinghtml/__init__.py
#   sphinxcontrib.qthelp (1.0.3) from /usr/lib/python3.8/site-packages/sphinxcontrib/qthelp/__init__.py
#   alabaster (0.7.12) from /usr/lib/python3.8/site-packages/alabaster/__init__.py
#   sphinx.ext.autodoc.preserve_defaults (1.0) from /usr/lib/python3.8/site-packages/sphinx/ext/autodoc/preserve_defaults.py
#   sphinx.ext.autodoc.type_comment (4.3.2) from /usr/lib/python3.8/site-packages/sphinx/ext/autodoc/type_comment.py
#   sphinx.ext.autodoc (4.3.2) from /usr/lib/python3.8/site-packages/sphinx/ext/autodoc/__init__.py
#   sphinx.ext.intersphinx (4.3.2) from /usr/lib/python3.8/site-packages/sphinx/ext/intersphinx.py
#   sphinx.ext.extlinks (4.3.2) from /usr/lib/python3.8/site-packages/sphinx/ext/extlinks.py
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/sphinx/setup_command.py", line 175, in run
    app.build(force_all=self.all_files)
  File "/usr/lib/python3.8/site-packages/sphinx/application.py", line 344, in build
    self.builder.build_update()
  File "/usr/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 291, in build_update
    self.build(['__all__'], to_build)
  File "/usr/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 358, in build
    self.write(docnames, list(updated_docnames), method)
  File "/usr/lib/python3.8/site-packages/sphinx/util/__init__.py", line 547, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/sphinx/builders/manpage.py", line 104, in write
    docwriter.write(largetree, destination)
  File "/usr/lib/python3.8/site-packages/docutils/writers/__init__.py", line 78, in write
    self.translate()
  File "/usr/lib/python3.8/site-packages/sphinx/writers/manpage.py", line 37, in translate
    visitor = self.builder.create_translator(self.document, self.builder)
  File "/usr/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 128, in create_translator
    return self.app.registry.create_translator(self, *args)
  File "/usr/lib/python3.8/site-packages/sphinx/registry.py", line 337, in create_translator
    translator = translator_class(*args)
  File "/usr/lib/python3.8/site-packages/sphinx/writers/manpage.py", line 102, in __init__
    self._docinfo['date'] = format_date(self.config.today_fmt or _('%b %d, %Y'),
  File "/usr/lib/python3.8/site-packages/sphinx/util/i18n.py", line 223, in format_date
    result.append(babel_format_date(date, babel_format, locale=language,
  File "/usr/lib/python3.8/site-packages/sphinx/util/i18n.py", line 187, in babel_format_date
    return formatter(date, format, locale=locale)
  File "/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/babel/dates.py", line 738, in format_datetime
    locale = Locale.parse(locale)
  File "/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/babel/core.py", line 302, in parse
    language = get_global('language_aliases').get(language, language)
  File "/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/babel/core.py", line 74, in get_global
    _raise_no_data_error()
  File "/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/babel/core.py", line 27, in _raise_no_data_error
    raise RuntimeError('The babel data files are not available. '
RuntimeError: The babel data files are not available. This usually happens because you are using a source checkout from Babel and you did not build the data files.  Just make sure to run "python setup.py import_cldr" before installing the library.
```

**Comment by akx** [comment]
The pep517 build problem should be a separate issue – it means the built package doesn't contain the pre-massaged data files.

**Comment by akx** [comment]
Sphinx issues fixed in #830. I broke out the PEP517 issue to #850.

Thank you!

**Comment by kloczek** [comment]
Tested my build with 5c6a9706 and 4f036e07 and still I see some warnings
```console
[tkloczko@devel-g2v SPECS]$ rpmbuild -ba  python-babel.spec
warning: Downloading https://github.com/python-babel/babel//archive/v2.9.1/python-babel-2.9.1.tar.gz to /home/tkloczko/rpmbuild/SOURCES/python-babel-2.9.1.tar.gz
warning: Downloading http://unicode.org/Public/cldr/37/core.zip#/python-babel-cldr-data.zip to /home/tkloczko/rpmbuild/SOURCES/python-babel-cldr-data.zip
warning: Downloading https://github.com/python-babel/babel//commit/5c6a9706.patch#/python-babel-Changelog-rename-to-.rst-use-gh-for-GitHub-links.patch to /home/tkloczko/rpmbuild/SOURCES/python-babel-Changelog-rename-to-.rst-use-gh-for-GitHub-links.patch
warning: Downloading https://github.com/python-babel/babel//commit/4f036e07.patch#/python-babel-Fix-Sphinx-indentation-complaint.patch to /home/tkloczko/rpmbuild/SOURCES/python-babel-Fix-Sphinx-indentation-complaint.patch
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.YaY4Tg
+ umask 022
+ cd /home/tkloczko/rpmbuild/BUILD
+ cd /home/tkloczko/rpmbuild/BUILD
+ rm -rf babel-2.9.1
+ /usr/bin/gzip -dc /home/tkloczko/rpmbuild/SOURCES/python-babel-2.9.1.tar.gz
+ /usr/bin/tar -xof -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd babel-2.9.1
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ /usr/bin/cat /home/tkloczko/rpmbuild/SOURCES/python-babel-Changelog-rename-to-.rst-use-gh-for-GitHub-links.patch
+ /usr/bin/patch -p1 -s --fuzz=0 --no-backup-if-mismatch -f
+ /usr/bin/cat /home/tkloczko/rpmbuild/SOURCES/python-babel-Fix-Sphinx-indentation-complaint.patch
+ /usr/bin/patch -p1 -s --fuzz=0 --no-backup-if-mismatch -f
+ /usr/bin/cp /home/tkloczko/rpmbuild/SOURCES/python-babel-cldr-data.zip cldr/cldr-core-37.zip
+ RPM_EC=0
++ jobs -p
+ exit 0

[..]

+ /usr/bin/sphinx-build -n -T -b man docs build/sphinx/man
Running Sphinx v4.5.0
making output directory... done
loading intersphinx inventory from https://docs.python.org/2/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: [new config] 27 added, 0 changed, 0 removed
reading sources... [100%] support
/usr/lib/python3.8/site-packages/babel/messages/extract.py:docstring of babel.messages.extract.extract_javascript:13: ERROR: Unexpected indentation.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... python-babel.3 { intro installation locale dates numbers messages cmdline setup support api/index api/core api/dates api/languages api/lists api/messages/index api/messages/catalog api/messages/extract api/messages/mofile api/messages/pofile api/numbers api/plural api/support api/units dev changelog license } /usr/lib/python3.8/site-packages/babel/dates.py:docstring of babel.dates.get_next_timezone_transition:1: WARNING: py:class reference target not found: TimezoneTransition
/usr/lib/python3.8/site-packages/babel/plural.py:docstring of babel.plural.PluralRule.parse:: WARNING: py:exc reference target not found: RuleError
/usr/lib/python3.8/site-packages/babel/plural.py:docstring of babel.plural.to_javascript:: WARNING: py:exc reference target not found: RuleError
/usr/lib/python3.8/site-packages/babel/plural.py:docstring of babel.plural.to_python:: WARNING: py:exc reference target not found: RuleError
/usr/lib/python3.8/site-packages/babel/plural.py:docstring of babel.plural.to_gettext:: WARNING: py:exc reference target not found: RuleError
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
/home/tkloczko/rpmbuild/BUILD/babel-2.9.1/docs/dates.rst:: WARNING: "table row spanning" not supported
done
build succeeded, 15 warnings.
```
You can probably ignore those warnings about `"table row spanning" not supported` because they are printed because  I'm generating man page and something is not fully supporting generate that kind of output.
